js/interface/map-panel.js

```diff
@@ -31,6 +74,41 @@ Fliplet.InteractiveMap.component('map-panel', {
       }
     },
     openMapPicker() {
+      this.dataSourceId = __widgetData[this.appId].data.markersDataSourceId
+
+      Fliplet.DataSources.connect(this.dataSourceId).then(connection => {
+        this.dataSourceConnection = connection
+        connection.find({where: {['Map name']: this.name}}).then(records => {
+          if (records.length) {
+            Fliplet.Modal.confirm({
+              title: 'Change image',
+              message: 'Do you want to keep the existing markers?',
+              buttons: {
+                confirm: {
+                  label: 'Keep the markers',
+                  className: 'btn-success'
+                },
+                cancel: {
+                  label: 'Delete the markers',
+                  className: 'btn-danger'
+                }
+              }
+            }).then(result => {
+              if (result) {
```

Comment:
as per code guidelines, return as early as possible to avoid indenting code:

```js
if (a) {
  // block 1
  return
}

// block 2
```

Reply:
Fixed